### ISSUE TITLE
Massive lag starting (any) Electron app.

### ISSUE THREAD

**Issue by SamEdgars** [electron/electron#11010]
Whenever I start my Electron apps (for the first time since starting the computer), it loads for several full seconds before the main.js window actually appears on the screen. From that point on, it's fast, and further starts of the Electron app are fast until I reboot the machine (or enough time passes?). My PC is very powerful and has a SSD disk, so it's not about my computer being slow.

While I suspect that nothing can be done about this, as it has to load in all the Chromium bloat, can't we at least give SOME sort of indication that something is actually being loaded in and soon will appear? Maybe a minimal "splash screen" that you can customize or something? I'm surprised that it's been so difficult so far to get anyone to understand what I mean. This must be happening for anyone who makes or runs Electron apps, no?

OS: Windows 10.

**Comment by felixrieseberg** [comment]
Hey Sam, a naked Electron application is actually very lightweight and requires less than 100mb in RAM  – which is half as much as, say, the `Apple Messages` app on macOS. Everything else is up to the people actually building the apps - displaying a splash screen, working hard on starting quickly, etc.

Chances are very high that something unique is going on with your computer. How long does Chrome take to launch on your computer? Is it also multiple seconds?

**Comment by SamEdgars** [comment]
I don't have Chrome installed and won't install it for security reasons. However, this has been the case on all the systems where I've tried any Electron app, on multiple different versions of Windows. So I'm extremely surprised to find that you don't even recognize this. Also, the claim that "a naked Electron application is actually very lightweight" makes little sense to me as it's a whole Chromium environment and I'm talking about even the example Electron app -- not something that does something exotic/weird/unusual.

**Comment by felixrieseberg** [comment]
How can we help you? A startup of multiple seconds for you is positively unique and I'm happy to help you figure that out, but it'll likely require some help (like, comparing the startup time with Chrome).

PS: "it's a whole Chromium environment" isn't accurate, but I understand that you likely just want stuff to be fast rather than learn about the innards of Electron.

**Comment by SamEdgars** [comment]
Since this is something that always happens for every Electron app, I think it's downright disturbing that you are denying this and acting as if it doesn't happen. Because it does, at least on Windows 10 and Windows 8.1, and regardless of how fast or slow the computer is. You must already be aware of this.

**Comment by Mike-Wood** [comment]
I saw your stack overflow post and thought I can explain in more detail here.

So when you create the mainWindow use `show: false ` in the options to stop it from showing immediately.

Then at the same time create a light weight splash page using `frame: false` and `transparent: true` and load a loading gif inside.

Then when the `did-finish-load` event is fired show the main window and hide the loading splash.
(If you look on the electron documentation it has a `ready-to-show` event but I found that it sometimes it fires too early or not at all).

[did-finish-load event](https://electron.atom.io/docs/api/web-contents/#event-did-finish-load)

[Frameless window docs](https://electron.atom.io/docs/api/frameless-window/#frameless-window) (further down it talks about transparent windows)

Here's a gif of what my one looks like:

![loadingspinner](https://user-images.githubusercontent.com/11504707/32404946-c26fe622-c174-11e7-984b-83bec46bf668.gif)
